app.js

```diff
@@ -1,60 +1,73 @@
-const harcodeJSON = ``
-
 const bodyParser = require('body-parser')
 const express = require('express')
 const app = express()
 
+const stations = require('./middleware/stations');
+
+
 
 // configure app to use bodyParser()
 // this will let us get the data from a POST
-app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.urlencoded({
+  extended: true
+}));
 app.use(bodyParser.json());
 
 app.get('/', (req, res) => {
-    res.send('Hello World!')
+  res.send('Hello World!')
+})
+
+app.get('/api/stations', (req, res) => {
+  console.log('fetch stations from original API');
+  stations.fetchStations()
+    .then((data) => {
+      res.status(200)
+      res.set('Content-Type', 'application/json')
+      res.json(data)
+    })
+    .catch((err) => console.log(err))
```

Comment:
To się nie może dostać na produkcję. Catch jest do obsługi błędów, nie ich diagnozowania. Po to masz chaining, że jak nie masz obsługi błędów <co jest ogólnie złe> to jej nie dodajesz do pipe'u

Reply:
Polecam
```
.catch(error) => next(error);
```

I zhandlowanie tego w dedykowanym middlewarze -> bardzo przyjemne rozwiązanie, można tworzyć własne obiekty błędów i mieć jeden handler do nich.

Nie zgodzę się w temacie obsługi błędu -> niezhandlowane odrzucenie Promise'a oznacza crash apki, a tego nie chcemy, więc już lepiej jest zalogować błąd, niż tego nie robić. A idealnie jest go obsłużyć (np. odsyłając 404 / 500)